examples/platform-switching/swift-platform/host.swift

```diff
@@ -1,62 +1,97 @@
-import Foundation
+// Build with latest Swift 6.0 using this command (adjusting for architecture):
+// swiftc -target aarch64-none-none-elf -Ounchecked -parse-as-library -enable-experimental-feature Embedded -wmo -c host.swift -o linux-arm64.o
 
-@_cdecl("roc_alloc")
-func rocAlloc(size: Int, _alignment: UInt) -> UInt  {
-    guard let ptr = malloc(size) else {
-        return 0
+@main
+enum Main {
+    static func main() {
+        var str = RocString()
+        mainForHost(&str)
+        
+        print(str, terminator: "")
     }
-    return UInt(bitPattern: ptr)
 }
 
-@_cdecl("roc_dealloc")
-func rocDealloc(ptr: UInt, _alignment: UInt)  {
-    free(UnsafeMutableRawPointer(bitPattern: ptr))
-}
+// Swift assumes its structs are not representable in C so the type has to be hidden
+// to get around a hardcoded safety check.
+@_extern(c, "roc__mainForHost_1_exposed_generic")
+func _mainForHost(_: UnsafeMutableRawPointer)
+func mainForHost(_ string: UnsafeMutablePointer<RocString>) { _mainForHost(UnsafeMutableRawPointer(string)) }
 
-@_cdecl("roc_realloc")
-func rocRealloc(ptr: UInt, _oldSize: Int, newSize: Int, _alignment: UInt) -> UInt {
-    guard let ptr = realloc(UnsafeMutableRawPointer(bitPattern: ptr), newSize) else {
-        return 0
+/// NOTE: The struct layout of Swift is that of C, without padding between fields (if alignment allows)
+/// There is no way to define this in a stable way yet.
+///
+/// Only the first three fields are stored to match RocStr memory layout
+struct RocString: BitwiseCopyable {
+    private let rawData: UnsafePointer<UInt8>?
+    private let rawCount: UInt
+    private let capacityOrAllocPointer: UInt // This is a union accessed through computed properties
+    
+    private var capacity: UInt { capacityOrAllocPointer }
+    private var allocPointer: UnsafeMutableRawPointer { .init(bitPattern: capacityOrAllocPointer)! }
+    
+    private var isSmallString: Bool { Int(bitPattern: capacity) < 0 }
+    
+    private var count: Int {
+        isSmallString // Small strings encode their count in the last byte of capacity
+            ? withUnsafeBytes(of: capacity) { bytes in Int(bytes.last! ^ 0b1000_0000) }
+            : Int(bitPattern: rawCount)
+    }
+    
+    // SAFETY: Not valid for small strings
+    private var data: UnsafeBufferPointer<UInt8> {
+        assert(isSmallString) // Assert this in debug builds
```

Comment:
Missed this, it should be `assert(!isSmallString)`, now it accidentally crashes correct code